### PR TITLE
can add product types

### LINKED ISSUE #7: As a client, I should be able to add a new product type
**Given** a client wants to create a new experimental product
**When** the client sends a POST request to /types
**Then** the API should add the type to the existing list
**And** respond with a JSON representation of the new type and a status code header of 201

**Given** a client wants to create a new experimental product
**When** the client sends a POST request to /types
**And** the description property is missing from the request body
**Then** the API should respond with an empty response body and a status code header of 400

### PR DESCRIPTION
**Pull request by sydneyrd**
Closes #7

#Description
Added create_product_type in product type view, added create product type to do_post

#Steps to test
Pull the sydney-add-product-type and switch to it
Start the API in debug mode
Open Postman and Post a product type 
   {
        "id": 4,
        "description": "test me"
    }
http://localhost:8088/product_types
Verify that you get a single dictionary back.
GET all product types and make sure the new type has been added

### CODE REVIEW

File: request_handler.py [hunk 1]
```diff
@@ -52,14 +52,16 @@ def do_GET(self):
...
         content_len = int(self.headers.get('content-length', 0))
         post_body = self.rfile.read(content_len)
-        response = { "payload": post_body }
-        self.wfile.write(json.dumps(response).encode())
+
+        post_body = json.loads(post_body)
+        (resource, id) = self.parse_url(self.path)
+        new_post = None
+        if resource == "product_types":
+            new_post = create_product_type(post_body)
```
Comment (stevebrownlee) [inline request_handler.py:62]: You are sending the request body to the creation function without doing any validation. This is one of the acceptance criteria in the #7 ticket. Please validate that the description property is in the request body before attempting to add to the list.

File: views/product_type_view.py [hunk 2]
```diff
@@ -25,3 +25,15 @@ def get_all_product_types():
...
+
+def create_product_type(product_type):
+    """Create a product type
+
+    Returns:
+        the new product type
+    """
+    max_id = PRODUCT_TYPES[-1]["id"]
+    new_id = max_id + 1
+    product_type["id"] = new_id
```
Comment (stevebrownlee) [inline views/product_type_view.py]: Would you consider combining these two operations into 1?

```py
product_type['id'] = max_id + 1
```

I don't believe that reduces the readability of the code.

**Review by stevebrownlee — CHANGES_REQUESTED** [review]

File: views/product_type_view.py (hunk 2 above)
Comment (sydneyrd) [reply views/product_type_view.py]: Yeah that looks better

File: request_handler.py (hunk 1 above)
Comment (sydneyrd) [reply request_handler.py:62]: Yep I fixed it!

**Review by stevebrownlee — APPROVED** [review]
Thanks for making the requested changes. Functionality works as expected.

**Pull request merged**